subsys/bluetooth/host/hci_core.c

```diff
@@ -845,6 +845,16 @@ static void hci_disconn_complete_prio(struct net_buf *buf)
 		return;
 	}
 
+	if (evt->reason == BT_HCI_ERR_CONN_FAIL_TO_ESTAB) {
+		/* No ACK or data was ever received. The peripheral may be
+		 * unaware of the connection attempt.
+		 *
+		 * Beware of confusing higher layer errors. Anything that looks
+		 * like it's from the remote is synthetic.
+		 */
+		LOG_WRN("conn 0x%04x failed to establish. RF noise?", handle);
```

Comment:
should you rather print the conn ptr? That's what we use for every other log in the stack.
You can also print the handle additionally ofc.

Reply:
If you look at the lines below, it appears to me we don't always have a live conn object in this situation. What's your advice?

Reply:
tentatively print it. if the upper layers are confused, then we must have a ref to it somewhere. If it's `nil` then the handle will help the devs piece back the puzzle

Reply:
> if the upper layers are confused, then we must have a ref to it somewhere

No, the conn object does not exist *yet* because of the event reordering madness. It will be created and destroyed after this handler has run.

I suppose I can move this warning to `bt_conn_set_state`, which always has the conn object.

Or we can do as you say and print the `null` here.

Any preference?

Reply:
alright how bout this
- change the log here to `conn handle [handle-hex] failed to establish.`
- add the conn handle to log messages in `bt_conn_set_state` in another commit
-

Reply:
I just realized I introduced a regression in https://github.com/zephyrproject-rtos/zephyr/commit/dd7caf667c8edd08e841bfb14f65ff96275d232c; it will lose the disconnect reason if the events are reordered.

Reply:
I moved the warning to `bt_conn_set_state`. It's called in `hci_disconn_complete_prio` if the conn exists, and will otherwise be called in `bt_hci_le_enh_conn_complete.